packages/azure-services/src/azure-batch/pool-load-generator.ts

```diff
@@ -95,10 +117,20 @@ export class PoolLoadGenerator {
 
         this.defaultActiveToRunningTasksRatio = configActiveToRunningTasksRatio;
         this.activeToRunningTasksRatio = configActiveToRunningTasksRatio;
-        this.timestamp = process.hrtime()[0];
+        this.samplingTimestamp = process.hrtime()[0];
     }
 
     private async getJobManagerConfig(): Promise<JobManagerConfig> {
         return this.serviceConfig.getConfigValue('jobManagerConfig');
     }
+
+    // tslint:disable: no-bitwise
+    private getActivityState(lastActivityState: number, isIdle: boolean): number {
+        const mask = 2147483647;
+        // shift previous states left and reset sign bit
+        const nextState = (lastActivityState << 1) & mask;
+
+        // set current state bit to 0/1 for idle/active pool respectively
+        return isIdle ? nextState : nextState + 1;
```

Comment:
why do we need this? can't we just deal with state as strings? like 'idle', 'running' etc?

Reply:
this is continuous pool states represented as bits